src/test/java/net/snowflake/client/jdbc/SnowflakeDriverLatestIT.java

```diff
@@ -735,4 +742,167 @@ private void testGeoMetadataSingle(
       }
     }
   }
+  /**
+   * Tests that upload and download small file to gcs stage. Require SNOW-206907 to be merged (still
+   * pass when not merge, but will use presigned url instead of GoogleCredential)
+   *
+   * @throws Throwable
+   */
+  @Test
```

Comment:
Can you refactor these tests a little to run under both the subscoped token and the presigned URL, and then pass in a flag or set the parameter to govern which one we use? We need to keep both code paths working for a while, so I want to make sure we have great coverage of both.

Reply:
The test for non subscoped is already in `SnowflakeDriverIT.java` (non latest). I did run that test already, so it have coverage for both.

Reply:
For regression, I did run t_stage locally and precommit, it did not have any issue. Make sense because the parameter default to false.

Reply:
Can we change the tests so that both the subscoped token tests and the presigned URL tests run through the same method so we're not duplicating code?

Reply:
We have Latest and non latest test for purpose of backward compatibility so I don't think that is a good idea.

Reply:
We can join into one when this is not "latest" anymore I think, not sure how jdbc team factoring the test code.

Reply:
What does "latest" mean?

Reply:
Based on the description of the file:

```
/**
 * General JDBC tests for the latest JDBC driver. This doesn't work for the oldest supported driver.
 * Revisit this tests whenever bumping up the oldest supported driver to examine if the tests still
 * is not applicable. If it is applicable, move tests to SnowflakeDriverIT so that both the latest
 * and oldest supported driver run the tests.
 */
```

Reply:
I see. Ok, thanks.